docs/Guide/preconditions/command-cooldown.mdx

```diff
@@ -1,6 +1,278 @@
+
+```javascript {7}
+const { Command } = require('@sapphire/framework');
+
+module.exports = class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      cooldownDelay: 10_000 // 10_000 milliseconds
+    });
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {7}
+import { Command } from '@sapphire/framework';
+
+export class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      cooldownDelay: 10_000 // 10_000 milliseconds
+    });
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {6}
+import { ApplyOptions } from '@sapphire/decorators';
+import { Command, CommandOptions } from '@sapphire/framework';
+
+@ApplyOptions<CommandOptions>({
+  // ...
+  cooldownDelay: 10_000 // 10_000 milliseconds
+})
+export class PingCommand extends Command {}
+```
+
+</TabItem>
+</Tabs>
+
+If you now try to run this command, and then run it again within 10 seconds, the command won't execute, and an error will
+be thrown. You can learn how to process that error [here][reporting-precondition-failure].
+
+:::info
+
+`cooldownDelay` only accepts a value of milliseconds, which is not known to be the easiest to read or calculate. To help,
+you can use [@sapphire/time-utilities], which includes utilities for time transformers.
+
+<Tabs groupId="language-choice">
+<TabItem value="javascript" label="JavaScript" default>
+
+```javascript {7}
+const { Command } = require('@sapphire/framework');
+const { Time } = require('@sapphire/time-utilities');
+
+module.exports = class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      cooldownDelay: Time.Second * 10; // Much easier for humans to parse
+    });
+  }
+};
+```
+
+</TabItem>
+<TabItem value="ESM" label="ESM">
+
+```javascript {7}
+import { Command } from '@sapphire/framework';
+import { Time } from '@sapphire/time-utilities';
+
+export class PingCommand extends Command {
+  constructor(context) {
+    super(context, {
+      // ...
+      cooldownDelay: Time.Second * 10; // Much easier for humans to parse
+    });
+  }
+}
+```
+
+</TabItem>
+<TabItem value="typescript" label="TypeScript">
+
+```typescript {6}
+import { ApplyOptions } from '@sapphire/decorators';
+import { Command, CommandOptions } from '@sapphire/framework';
+import { Time } from '@sapphire/time-utilities';
+
+@ApplyOptions<CommandOptions>({
+  // ...
+  cooldownDelay: Time.Second * 10; // Much easier for humans to parse
+})
+export class PingCommand extends Command {}
+```
+
+</TabItem>
+</Tabs>
+
+You can view the docs [here][timeutils].
+
+:::
+
+## User Exceptions
+
+It's very common to not want cooldowns to apply to certain people, for example, the bot owner. This can be achieved by
+adding [`cooldownFilteredUsers`][cooldownfilteredusers] to the options. This option should be an array of users ID that
```

Comment:
Unrelated to your PR but oh god its a hardcoded array? who thought that was a good idea? .w.
Can it be a callback too? If yes, show that too please!

Reply:
It's in the constructor in which you can do anything above making the super call

Reply:
Yeah but you can only access the `context` for that, right? If it's a function you could give it the `message` argument to, for example, query the db, grab the mod role, and ignore all mods.

Reply:
Uh yeah that's true. This'll be something to pick up at a later date when we can figure out a good way to go about this I think. The precondition itself gets the message of course so that's a starting point at least.